MIGRATION_GUIDE.md

```diff
@@ -38,11 +38,54 @@ As part of removing legacy APIs or authentication flows no longer recommended fo
 
 Continue reading for the detail of classes and methods that were impacted.
 
+### Updated Callbacks
+The widget requires a callback to receive the results in. The interface for this is `LockCallback`, which takes either an event or an error. The `onError` method got updated to receive an `AuthenticationException` instead of `LockException`. This change will help developers extract the *code* and *description* of the error and understand better what went wrong and how to recover from it.
+
+The change impacts the abstract subclass `AuthenticationCallback`. Additionally, this class no longer has an `onCanceled` method. If you need to handle this scenario you have two options:
+- Implement `LockCallback` and handle the different event types, checking for `LockEvent.CANCELED`.
+- Implement `AuthenticationCallback` and check the received exception for the property `AuthenticationException#isCanceled()`.
+
+```kotlin
+// Before
+val callback: LockCallback = object : AuthenticationCallback() {
+    override fun onAuthentication(credentials: Credentials) {
+        // Authenticated
+    }
+
+    override fun onCanceled() {
+        // Canceled
+    }
+
+    override fun onError(error: LockException) {
+        // Another error. Check code & description.
+    }
+}
+
+// After
+val callback: LockCallback = object : AuthenticationCallback() {
+    override fun onAuthentication(credentials: Credentials) {
+        // Authenticated
+    }
+
+    override fun onError(error: AuthenticationException) {
+        if (error.isCanceled) {
+            // Canceled
+        } else {
+            // Another error. Check code & description.
+        }
+    }
+}
+```
+
 ### Removed classes
 - `VoidCallback` is no longer available. Please, use `Callback<Void, AuthenticationException>` instead.
+- `LockException` is no longer available. This impacts the `LockCallback` and `AuthenticationCallback` classes. Please, use `AuthenticationException` instead.
 
 ### Removed methods
 
+#### From class `AuthenticationCallback`
+- Removed `public void onCanceled()`. Instead, an exception will be raised with this code. Check the property `AuthenticationException#isCanceled()` to handle it.
```

Comment:
Says "with this code" but doesn't include the code.

Reply:
Good catch. Rephrased this and also noted that this is a method and not a property. Ready for review.